GammaGeeks/tura-backend
# PR Description: Implement "Sign Up with Google" Button

### PR Description: Implement "Sign Up with Google" Button  

**What does this PR do?**  
This PR introduces the functionality to allow new users to sign up for the Tura application using their Google accounts. This provides a quick and seamless registration process by leveraging Google’s authentication services to handle account creation and login.

**Description of the task to be completed:**  
- Add a "Sign Up with Google" button on the sign-up page.  
- Implement Google OAuth to prompt users to authorize the application for account creation.  
- Automatically create a new user account using profile information (e.g., name, email) retrieved from Google.  
- Log in the user after successful account creation and redirect them to the user dashboard.  
- Display an error message if the Google authorization process fails.  
- Securely manage Google authentication tokens and sessions.  
- Ensure first-time social sign-ups create new entries in the database.

**How should this be manually tested?**  
1. Clone the repository and navigate to the project directory.  
2. Install all dependencies and start the backend server.  
3. Navigate to the sign-up page and click the "Sign Up with Google" button.  
4. Complete the Google authorization process and verify account creation and redirection to the dashboard.  
5. Simulate an authorization failure (e.g., by denying access) and confirm that an error message is displayed.  
6. Verify that tokens are securely handled and user information is saved correctly in the database for first-time sign-ups.

**What are the relevant pivotal tracker stories?**  
- **Title:** User Sign-Up via Google  
- **IDs:**  [#188511353](https://www.pivotaltracker.com/n/projects/2725084/stories/188511353) , [#188511356](https://www.pivotaltracker.com/n/projects/2725084/stories/188511356)
